### PR TITLE
jms: cleans up artemis code in attempts to deflake test

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This extracts code I started to refactor while trying to get to the bottom of [this error on an unrelated PR](https://github.com/openzipkin/brave/actions/runs/7431666507).

```
 Error: [ERROR] brave.jakarta.jms.ITJmsTracingMessageConsumer.receive_customSampler -- Time elapsed: 0.028 s <<< ERROR!
[INFO] jakarta.jms.JMSException: AMQ219010: Connection is destroyed
[INFO] 	at org.apache.activemq.artemis.core.protocol.core.impl.ChannelImpl.sendBlocking(ChannelImpl.java:478)
[INFO] 	at org.apache.activemq.artemis.core.protocol.core.impl.ChannelImpl.sendBlocking(ChannelImpl.java:452)
[INFO] 	at org.apache.activemq.artemis.core.protocol.core.impl.ActiveMQClientProtocolManager.createSessionContext(ActiveMQClientProtocolManager.java:308)
[INFO] 	at org.apache.activemq.artemis.core.protocol.core.impl.ActiveMQClientProtocolManager.createSessionContext(ActiveMQClientProtocolManager.java:256)
[INFO] 	at org.apache.activemq.artemis.core.client.impl.ClientSessionFactoryImpl.createSessionChannel(ClientSessionFactoryImpl.java:1523)
[INFO] 	at org.apache.activemq.artemis.core.client.impl.ClientSessionFactoryImpl.createSessionInternal(ClientSessionFactoryImpl.java:837)
[INFO] 	at org.apache.activemq.artemis.core.client.impl.ClientSessionFactoryImpl.createSession(ClientSessionFactoryImpl.java:341)
[INFO] 	at org.apache.activemq.artemis.jms.client.ActiveMQConnection.authorize(ActiveMQConnection.java:650)
[INFO] 	at org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory.createConnectionInternal(ActiveMQConnectionFactory.java:925)
[INFO] 	at org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory.createConnection(ActiveMQConnectionFactory.java:290)
[INFO] 	at org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory.createConnection(ActiveMQConnectionFactory.java:285)
[INFO] 	at brave.jakarta.jms.ArtemisJmsExtension.newConnection(ArtemisJmsExtension.java:69)
[INFO] 	at brave.jakarta.jms.JmsExtension.beforeEach(JmsExtension.java:82)
[INFO] 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
[INFO] 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
[INFO] Caused by: ActiveMQNotConnectedException[errorType=NOT_CONNECTED message=AMQ219010: Connection is destroyed]
[INFO] 	... 15 more
[INFO] 
```

@reta this is on a shared branch, if you can see anything we can do to avoid flakes in activemq, please give a try. Note the failure was in the jarkarta invoker tests, and possibly resource related in CI. If there's a way to make things leaner maybe it will pass. Also, if there'a another MQ we don't need to stick with artemis either. In zipkin and reporter, I moved to docker for reasons like this. However, since this is instrumenting, I'm not sure if we can or not.. I guess we could as we arent' instrumenting the broker... Anyway, if you have some time to give a try, please do as this flake is locking up the project. If we can use docker, yank the ActiveMQExtension from zipkin or reporter and swap it out!

### CODE REVIEW

File: instrumentation/jms-jakarta/src/test/java/brave/jakarta/jms/ITTracingJMSConsumer.java [hunk 1]
```diff
@@ -40,7 +40,7 @@
  * {@link brave.jakarta.jms.ITTracingMessageConsumer}
  */
 class ITTracingJMSConsumer extends ITJms {
-  @RegisterExtension ArtemisJmsExtension jms = new ArtemisJmsExtension();
+  @RegisterExtension JmsExtension jms = new JmsExtension();
```
Comment (codefromthecrypt) [inline instrumentation/jms-jakarta/src/test/java/brave/jakarta/jms/ITTracingJMSConsumer.java:43]: the non-artemis one was dead code in this project

File: instrumentation/jms-jakarta/src/test/java/brave/jakarta/jms/JmsExtension.java [hunk 2]
```diff
@@ -104,41 +157,16 @@ abstract void setReadOnlyProperties(Message message, boolean readOnlyProperties)
...
+      if (topicSession != null) topicSession.close();
+      if (topicConnection != null) topicConnection.close();
+      if (queueSession != null) queueSession.close();
+      if (queueConnection != null) queueConnection.close();
     } catch (JMSException e) {
       throw new AssertionError(e);
     }
-  }
-
-  static class ActiveMQ extends JmsExtension {
```
Comment (codefromthecrypt) [inline instrumentation/jms-jakarta/src/test/java/brave/jakarta/jms/JmsExtension.java:118]: this was dead code so I flattened the type.

File: instrumentation/jms/src/test/java/brave/jms/JmsExtension.java [hunk 3]
```diff
@@ -100,12 +100,12 @@ abstract void setReadOnlyProperties(Message message, boolean readOnlyProperties)
 
   @Override public void afterEach(ExtensionContext context) throws Exception {
     try {
-      session.close();
-      connection.close();
-      topicSession.close();
-      topicConnection.close();
-      queueSession.close();
-      queueConnection.close();
+      if (session != null) session.close();
```
Comment (codefromthecrypt) [inline instrumentation/jms/src/test/java/brave/jms/JmsExtension.java:103]: when the first flake happened, it was obscured by a NPE, so I fixed in both places

**Review by codefromthecrypt — COMMENTED** [review]
here are some notes in case helps. If you don't have time, I can swing the axe again tomorrow.

here are the places I switched to docker due to flakey embedded servers

https://github.com/openzipkin/zipkin/tree/master/zipkin-collector/activemq/src/test/java/zipkin2/collector/activemq
https://github.com/openzipkin/zipkin-reporter-java/tree/master/activemq-client/src/test/java/zipkin2/reporter/activemq

**Comment by codefromthecrypt** [comment]
also, for some reason, in the other PR only the JDK11 run of the jms-jakarta invoker test was flaking, 21 hasn't so far.

**Comment by reta** [comment]
> also, for some reason, in the other PR only the JDK11 run of the jms-jakarta invoker test was flaking, 21 hasn't so far.

This is very interesting ... shouldn't be JDK dependent ...

File: instrumentation/jms-jakarta/src/it/jms30/pom.xml [hunk 4]
```diff
@@ -20,13 +20,14 @@
   <modelVersion>4.0.0</modelVersion>
 
   <groupId>@project.groupId@</groupId>
-  <artifactId>jms40</artifactId>
```
Comment (reta) [inline instrumentation/jms-jakarta/src/it/jms30/pom.xml:23]: Sorry about that, it was definitely my mistake

File: instrumentation/jms-jakarta/src/test/java/brave/jakarta/jms/JmsExtension.java [hunk 5]
```diff
@@ -29,16 +30,28 @@
...
+import org.apache.activemq.artemis.core.settings.impl.AddressSettings;
+import org.apache.activemq.artemis.jms.client.ActiveMQJMSConnectionFactory;
 import org.apache.activemq.artemis.jms.client.ActiveMQMessage;
 import org.junit.jupiter.api.extension.AfterEachCallback;
 import org.junit.jupiter.api.extension.BeforeEachCallback;
 import org.junit.jupiter.api.extension.ExtensionContext;
 
-public abstract class JmsExtension implements BeforeEachCallback, AfterEachCallback {
+/**
+ * ActiveMQ 6 supports JMS 2.0, but requires JDK 17. We have to build on JDK
```
Comment (reta) [inline instrumentation/jms-jakarta/src/test/java/brave/jakarta/jms/JmsExtension.java:47]: 👍 was also aware of that but JDK-17 is indeed a hard requirement

**Comment by reta** [comment]
> here are some notes in case helps. If you don't have time, I can swing the axe again tomorrow.

@codefromthecrypt happy to look into that (it may took me a few days), we have fought a few flaky tests in Apache CXF related to Artemis migration

**Review by reta — APPROVED** [review]
@codefromthecrypt I was running the tests (from this pull request) for at least an hour continuously and no flaky failures observed so far, this is good cleanup in any case to be merged, thanks!

**Comment by codefromthecrypt** [comment]
@reta agree to merge this. I'm wondering if CI was having a bad day. In any case, we can revisit later as isn't currently a blocker.

**Comment by codefromthecrypt** [comment]
opened an issue as the flake returned https://github.com/openzipkin/brave/actions/runs/7438289752/job/20236893811?pr=1395

https://github.com/openzipkin/brave/issues/1398

**Pull request merged**